recipes/mapd-core-cpu/build.sh

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env bash
+
+set -ex
+
+env
+
+ls -la $BUILD_PREFIX/bin
+
+# sed -i option is handled differently in Linux and OSX
+if [ $(uname) == Darwin ]; then
+    INPLACE_SED="sed -i \"\" -e"
+else
+    INPLACE_SED="sed -i"
+fi
+
+# mapd-core v 4.5.0 (or older) hardcodes /usr/bin/java. Grab
+# Calcite.cpp with a fix
+# (https://github.com/omnisci/mapd-core/pull/316) from a repo:
+wget https://raw.githubusercontent.com/omnisci/mapd-core/7c1faa09dd88d0cc735b629048f74d71baa9179f/Calcite/Calcite.cpp
+mv Calcite.cpp Calcite/
+
+# conda build cannot find boost libraries from
+# ThirdParty/lib. Actually, moving environment boost libraries to
+# ThirdParty/lib does not make much sense. The following is just a
+# quick workaround of the problem. Upstream will remove the relevant
+# code from CMakeLists.txt as not needed.
+$INPLACE_SED 's/DESTINATION ThirdParty\/lib/DESTINATION lib/g' CMakeLists.txt
+
+export LDFLAGS="-L$PREFIX/lib -Wl,-rpath,$PREFIX/lib"
+
+# Enforce PREFIX instead of BUILD_PREFIX:
+export ZLIB_ROOT=$PREFIX
+export LibArchive_ROOT=$PREFIX
+export Curses_ROOT=$PREFIX
+
+if [ $(uname) == Darwin ]; then
+    # Darwin has only clang
+    COMPILERNAME=clang   # options: clang
+    export CC=$CLANG
+    export CXX=$CLANGXX
+else
+    # Linux
+    echo "uname=${uname}"
+    COMPILERNAME=clang   # options: clang, gcc
+    GXX=$BUILD_PREFIX/bin/$HOST-g++  # replace with $GXX
+    GCCSYSROOT=$BUILD_PREFIX/$HOST/sysroot
+    GCCVERSION=$(basename $(dirname $($GXX -print-libgcc-file-name)))
+    GXXINCLUDEDIR=$BUILD_PREFIX/$HOST/include/c++/$GCCVERSION
+    GCCLIBDIR=$BUILD_PREFIX/lib/gcc/$HOST/$GCCVERSION
+
+    echo "GCCVERSION=$GCCVERSION"
+    echo "GCCSYSROOT=$GCCSYSROOT"
+    echo "GCCLIBDIR=$GCCLIBDIR"
+    echo "GXXINCLUDEDIR=$GXXINCLUDEDIR"
+    
+    # Fix not found include file errors:
+    CXXINC1=$GXXINCLUDEDIR            # cassert, ...
+    CXXINC2=$GXXINCLUDEDIR/$HOST      # <string> requires bits/c++config.h
+    CXXINC3=$GCCSYSROOT/usr/include   # pthread.h
+    
+    # Add include directories for explicit clang++ call in
+    # QueryEngine/CMakeLists.txt for building RuntimeFunctions.bc and
+    # ExtensionFunctions.ast:
+    mv QueryEngine/CMakeLists.txt QueryEngine/CMakeLists.txt-orig
+    echo -e "set(CXXINC1 \"-I$CXXINC1\")" > QueryEngine/CMakeLists.txt
+    echo -e "set(CXXINC2 \"-I$CXXINC2\")" >> QueryEngine/CMakeLists.txt
+    echo -e "set(CXXINC3 \"-I$CXXINC3\")" >> QueryEngine/CMakeLists.txt
+    cat QueryEngine/CMakeLists.txt-orig >> QueryEngine/CMakeLists.txt
```

Comment:
Maybe a patch with these changes is more clear than changing it in the build script, no?

Reply:
Waiting for feedback from upstream on this.